src/modules/AuthService.test.ts

```diff
@@ -0,0 +1,50 @@
+import * as AuthService from "src/modules/AuthService";
+import { User } from "src/modules/User";
+
+jest.mock("firebase", () => ({
+  initializeApp: jest.fn(() => {
+    const functions = require("firebase-functions-test");
+    const testEnv = functions();
+    const firebaseUser = testEnv.auth.exampleUserRecord();
+    return {
+      auth: () => {
+        return {
+          signInWithEmailAndPassword: () =>
+            Promise.resolve({ user: firebaseUser }),
+          createUserWithEmailAndPassword: () =>
+            Promise.resolve({ user: firebaseUser }),
+          sendPasswordResetEmail: () => Promise.resolve(),
+          signOut: () => Promise.resolve(),
+        };
+      },
+    };
+  }),
+}));
+
+describe("Auth Service", () => {
```

Comment:
> Can you add some tests for failure scenarios

Reply:
Unable to add fail scenrios in same file since I mocked it to have a success case will have to make different test file for that couldn't mock different different return values for nested objects that contains function.